cmk/base/plugins/agent_based/cisco_asa_sensors.py

```diff
@@ -0,0 +1,340 @@
+        if sensorname != '':  # for asa context, there are no real sensors.
+            if sensortype == '8':  # Temperature
+                sensors['temp'].update({sensorname.replace('Temperature ', ''): CiscoAsaSensor(
+                    value=to_celsius(float(sensorvalue), sensorunits),
+                    unit=sensorunits,
+                    status=get_sensor_status(sensorstatus),
+                    state_readable=get_state_readable(sensorstatus),
+                )})
+
+            if sensortype == '10':  # Fan
+                sensors['fan'].update({sensorname.replace('Fan ', ''): CiscoAsaSensor(
+                    value=int(sensorvalue),
+                    unit=sensorunits,
+                    status=get_sensor_status(sensorstatus),
+                    state_readable=get_state_readable(sensorstatus),
+                )})
+
+            if sensortype == '12':  # Power supply
+                sensors['power'].update({sensorname.replace('Power ', ''): CiscoAsaSensor(
+                    value=0,
+                    unit='',
+                    status=get_sensor_status(sensorstatus),
+                    state_readable=get_state_readable(sensorstatus),
+                )})
+
+    return sensors
+
+
+register.snmp_section(
+    name='cisco_asa_sensors',
+    parse_function=parse_cisco_asa_sensors,
+    fetch=[
+        SNMPTree(
+            base='.1.3.6.1.2.1',  #
+            oids=[
+                '47.1.1.1.1.7',  # ENTITY-MIB::entPhysicalName
+                '99.1.1.1.1',  # ENTITY-SENSOR-MIB::entPhySensorType
+                '99.1.1.1.4',  # ENTITY-SENSOR-MIB::entPhySensorValue
+                '99.1.1.1.5',  # ENTITY-SENSOR-MIB::entPhySensorOperStatus
+                '99.1.1.1.6',  # ENTITY-SENSOR-MIB::entPhySensorUnitsDisplay
+            ]
+        ),
+    ],
+    detect=startswith('.1.3.6.1.2.1.1.1.0', 'cisco adaptive security appliance')
+)
+
+# ##################################################################################################
+#
+# ASA SENSORS TEMPERATURE
+#
+# ##################################################################################################
+
+from .utils.temperature import (
+    check_temperature,
+    TempParamType,
+    to_celsius,
+)
+
+
+def discovery_cisco_asa_temp(section: Dict) -> DiscoveryResult:
+    for key in section['temp']:
+        yield Service(item=key)
+
+
+def check_cisco_asa_temp(item, params: TempParamType, section) -> CheckResult:
+    try:
+        sensor = section['temp'][item]
+
+        yield Result(state=sensor.status, summary='Status: %s' % sensor.state_readable)
+
+        yield from check_temperature(
+            sensor.value,
+            dev_unit=sensor.unit,
+            dev_status=sensor.status,
+            dev_status_name=sensor.state_readable,
+            params=params,
+            unique_name='check_cisco_asa_temp.%s' % item,
+        )
+    except KeyError:
```

Comment:
Please move the except clause further up, right after `sensor = ...` . You can use `return` here, the Check-API supports this.